query-service/api.raml

```diff
@@ -2,293 +2,400 @@
-        description: |
-          Internal Server Error
-          
-          An unexpected exception was thrown while attempting to process
-          the request.
+        description: Server Error
         body:
           application/json:
             type: error.ServerError
 
-# TODO: need the file upload patch in the gradle plugin for this
-/word-count:
   post:
-    description: Counts the words in an uploaded file.
+    displayName: Create New Job
+    description: Creates a new blast query job.
     body:
       multipart/form-data:
-        type: file
+        type: object
+        additionalProperties: false
+        properties:
+          config:
+            displayName: Job Configuration
+            type: lib.QueryJobRequest
+            required: true
+          query:
+            displayName: Query File
+            description: |
+              Optional file upload parameter for uploading multi-blast queries.
+              
+              **IMPORTANT**: Either this field or the job config's `query` field
+              MUST be set in order for a job to be valid.
+            type: file
+            required: false
     responses:
       200:
         description: |
-          Success
-          
-          Job was processed and queued successfully.
+          Query job was successfully submitted to the job queue for execution.
+        body:
+          application/json:
+            type: lib.QueryJobDetails
+      400:
+        description: |
+          Invalid request body or parameters.
+        body:
+          application/json:
+            type: error.BadRequestError
+      401:
+        description: |
+          No valid user authentication provided.
+        body:
+          application/json:
+            type: error.UnauthorizedError
+      422:
+        description: |
+          Invalid job configuration.
+        body:
+          application/json:
+            type: error.UnprocessableEntityError
+
+
+# ---------------------------------------------------------------------------- #
+#                                                                              #
+#   Job By ID                                                                  #
+#                                                                              #
+# ---------------------------------------------------------------------------- #
+/jobs/{job-id}:
+  displayName: Job By ID
+  description: |
+    Operations on a singular job identified by the `{job-id}` URI parameter.
+
+  uriParameters:
+    job-id:
+      type: string
+      pattern: ^[\dA-Fa-f]{32}$
+      minLength: 32
+      maxLength: 32
+
+  get:
+    displayName: Get Job Details
+    description: |
+      Returns details about the target job.
+    
+      As a side effect, interactions with this endpoint will cause the target
+      job to be added to the requesting user's job collection if it was not
+      already present in that user's collection.  This default behavior can be
```

Comment:
Mention in this comment that the use case is sharing.

Reply:
Updated